packages/workspace/src/task-runner/task-runner.ts

```diff
@@ -0,0 +1,28 @@
+import { Observable } from 'rxjs';
+import { Target } from '@angular-devkit/architect';
+
+export interface Task {
+  id: string;
+  target: Target;
+  overrides: Object;
+  dependencies: string[];
+}
+
+export enum BuildEventType {
+  TaskComplete = '[Task] Complete'
+}
+
+export interface BuildEvent {
+  task: Task;
+  type: BuildEventType;
+}
+
+export interface TaskCompleteEvent extends BuildEvent {
+  type: BuildEventType.TaskComplete;
+  success: boolean;
+}
+
+export type TasksRunner = (
+  tasks: Task[],
```

Comment:
Given my comments about the graph, we might change it to an object that looks like this:

```
{
  tasks: {id: string; target: Target, overrides}
  roots: string[]
  deps: {[id: string]: string[]}
}
```